### PR TITLE
Fix create-test-db command

### LINKED ISSUE #24547: Unable to run `composer create-test-db`
It is not possible to run Behat test on develop or in 1.7.8.x. 
Tests need a database to be run on. According to docs,  `composer create-test-db` is used to create test database.
But this command fails.

#### Expected behavior

Tests to be run.

#### Steps to Reproduce

1. Go to develop and pull the latest changes
2. run  `composer create-test-db`
3. See Error: Fatal error:  Uncaught PrestaShopException: Property Profile->name is empty

#### Additional information

* PrestaShop branches: develop, 1.7.8.x 
* PHP versions: 7.3, 7.2

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Some database tables were missing, because doctrine wasn't loaded.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | yes - rename method `src/PrestaShopBundle/Install/Install.php::generateSf2ProductionEnv()` -> `updateSchema()`.
| Deprecations?     | no
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/24547
| How to test?      | run composer create-test-db and no errors + test_prestashop database created locally.
| Possible impacts? | Need to double check if all prestashop installation methods are still working (install through web `/install-dev` and console command (`/install-dev/index_cli`)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24711)

### CODE REVIEW

File: src/PrestaShopBundle/Install/Install.php [hunk 1]
```diff
@@ -311,22 +311,18 @@ public function installDatabase($clear_database = false)
...
      * @return bool
      */
-    public function generateSf2ProductionEnv()
+    public function generateSf2Env()
     {
-        if (defined('_PS_IN_TEST_')) {
-            return true;
-        }
         $schemaUpgrade = new UpgradeDatabase();
         $schemaUpgrade->addDoctrineSchemaUpdate();
```
Comment (zuk3975) [inline src/PrestaShopBundle/Install/Install.php:327]: if you check inside this method it runs the command with --env=PS_ENV, so it should work as expected in TEST env too.

File: src/PrestaShopBundle/Install/Install.php [hunk 2]
```diff
@@ -311,22 +311,18 @@ public function installDatabase($clear_database = false)
...
     /**
-     * Pass SF2 to production
      * cache:clear
      * assetic:dump
      * doctrine:schema:update.
      *
      * @return bool
      */
-    public function generateSf2ProductionEnv()
+    public function generateSf2Env()
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Install/Install.php]: This is a BC break

Comment (zuk3975) [reply src/PrestaShopBundle/Install/Install.php]: Right, its not necessary though, just method doesnt seem to do anything specific to production though. Do you think i should leave original name?

Comment (zuk3975) [reply src/PrestaShopBundle/Install/Install.php]: ok I've left the original name. But if you agree to BC break, I guess It should at least be more meaningful.
Smth like `updateSchema()` ?. Let me know what you think.

Comment (PierreRambaud) [reply src/PrestaShopBundle/Install/Install.php]: I totally agree about the BC break, it was about adding it in the PR's description ^^'
` updateSchema`  seems to be a very good idea

**Comment by mvorisek** [comment]
This is already fixed in https://github.com/PrestaShop/PrestaShop/pull/22823

**Review by PierreRambaud — DISMISSED** [review]

**Comment by mvorisek** [comment]
@PierreRambaud did you read https://github.com/PrestaShop/PrestaShop/pull/22823#issuecomment-847747665 and tested that PR?

**Comment by PierreRambaud** [comment]
> @PierreRambaud did you read [#22823 (comment)](https://github.com/PrestaShop/PrestaShop/pull/22823#issuecomment-847747665) and tested that PR?

No, because I'm still convinced it's not the right way to do it, you remove parts of code that I was against on.

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by mvorisek** [comment]
@PierreRambaud I know, but the removed part is the only way to unify SF/non-SF config. And if you will test it, you will find out, that the removed prefixing part is working. Please test.

**Comment by matthieu-rolland** [comment]
I guess this PR should be tested by a dev instead of the QA team ?  (I put the Waiting for QA label so that QA can confirm they won't test this PR)

**Comment by Progi1984** [comment]
> I guess this PR should be tested by a dev instead of the QA team ? (I put the Waiting for QA label so that QA can confirm they won't test this PR)

@matthieu-rolland We need to ask to @PrestaShop/qa-functional ;)

**Comment by Progi1984** [comment]
@zuk3975 Could you target the 178x branch for this PR, please ?

**Comment by Robin-Fischer-PS** [comment]
Hi there,

QA will pass some tests, but if the dev can test the more technical part (CLI install) it would be nice :)

**Comment by mvorisek** [comment]
@Robin-Fischer-PS please test also https://github.com/PrestaShop/PrestaShop/pull/22823

**Comment by zuk3975** [comment]
> @zuk3975 Could you target the 178x branch for this PR, please ?

okay, but it means I need to close this PR and open a new one branched from 1.7.8.x

**Comment by zuk3975** [comment]
moved same changes to 178x target https://github.com/PrestaShop/PrestaShop/pull/24736

**Pull request closed without merging**